Content.Server/Medical/HealingSystem.cs

```diff
@@ -159,6 +166,25 @@ private bool TryHeal(EntityUid uid, EntityUid user, EntityUid target, HealingCom
         return true;
     }
 
+    /// <summary>
+    /// Scales the self-heal penalty based on the amount of damage taken
+    /// </summary>
+    /// <param name="uid"></param>
+    /// <param name="component"></param>
+    /// <returns></returns>
+    public float GetScaledHealingPenalty(EntityUid uid, HealingComponent component)
+    {
+        var output = component.Delay;
+        if (!TryComp<MobStateComponent>(uid, out var mobState) || !TryComp<DamageableComponent>(uid, out var damageable))
+            return output;
+
+        _mobStateSystem.TryGetEarliestCriticalState(mobState, 0, out var _, out var amount);
+        var percentDamage = (float) (damageable.TotalDamage / amount);
+        //basically make it scale from 1 to the multiplier.
+        var modifier = percentDamage * (component.SelfHealPenaltyMultiplier - 1) + 1;
+        return Math.Max(modifier, 0);
```

Comment:
```suggestion
        return Math.Max(modifier, 1);
```

If < 1 is a possibility, then the delay multiplier could make the delay shorter than it's supposed to be (since this is supposed to be a penalty, that might not be accurate)